client/tinychain/math/operator.py

```diff
@@ -78,7 +78,7 @@ def backward(self, variable):
 
     def gradients(self, loss):
```

Comment:
An `Adam` optimizer needs to know the namespace of its model's `Variable`s at `__init__` time, which means that a `Variable` which is a transform of another `Variable` needs a method to transform a given `loss` to match the shape of the source `Variable`. You have a point though that `gradients` is maybe not a good name for this because it doesn't actually calculate a gradient, it only inverts a transform. I renamed it to `invert`.